vyper/ir/optimizer.py

```diff
@@ -662,10 +662,10 @@ def _rewrite_mstore_dload(argz):
 def _merge_mload(argz):
     if not version_check(begin="cancun"):
         return False
-    return _merge_load(argz, "mload", "mcopy")
+    return _merge_load(argz, "mload", "mcopy", allow_overlap=False)
 
 
-def _merge_load(argz, _LOAD, _COPY):
+def _merge_load(argz, _LOAD, _COPY, allow_overlap=True):
```

Comment:
nit: think generally params should be "safe-by-default", and the "safer" option would be to have no overlap.

Reply:
hmm i see, my thinking was that this only affects where the source and destination address spaces are the same, and there is only one case for this, `mload` / `mstore`. but i am open to changing it.